hpy/devel/include/common/runtime/ctx_type.h

```diff
@@ -0,0 +1,18 @@
+#ifndef HPY_COMMON_RUNTIME_CTX_TYPE_H
+#define HPY_COMMON_RUNTIME_CTX_TYPE_H
+
+#include <Python.h>
```

Comment:
Why do we include Python.h here?

Reply:
Because this is the header file corresponding to `ctx_type.c`, which is used in two places:
1. in the CPython-ABI mode (in this case we are obviously allowed to include Python.h)
2. to implement `hpy.universal` for CPython (and we are again allowed to include Python.h)

Note that this file is **not** included nor used if you just `#include <hpy.h>` in universal mode.
However, I agree that it's confusing to have it under `hpy/devel/include/common`: probably this means that we should restructure the layout in a way which makes more clear what is the interface and what is the implementation.

Maybe something like:
1. `hpy/devel/include`, which contains everything needed to compile an universal extension. No `Python.h` allowed there
2. `hpy/devel/src` (or `common`, `runtime`, `cpython`, ...) which contains implementation files which are used **both** by `hpy/devel/include/cpython` and `hpy/universal/src/` ?